mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -744,7 +744,13 @@ private[classification] class MultiClassSummarizer extends Serializable {
   def countInvalid: Long = totalInvalidCnt
 
   /** @return The number of distinct labels in the input dataset. */
-  def numClasses: Int = distinctMap.keySet.max + 1
+  def numClasses: Int = try {
+    distinctMap.keySet.max + 1
+  } catch {
+    case e: UnsupportedOperationException =>
```

Comment:
Is the only result of returning 0 that histogram() returns an empty array? that seems ... fine. It's not meaningful but it's not wrong.